### PR TITLE
Update CONTRIBUTING with more workflow info

### PR DESCRIPTION
**Pull request by humphd**
I've seen people get a bunch of this wrong, and wanted to try and help them get the workflows right by adding better docs.

This updates our CONTRIBUTING docs to include:

- workflow for git and github
- more clarity on how to run tests, fix things with eslint
- dealing with updates in `package.json` and `package-lock.json`

See if you think I've covered everything.  If there are other things I didn't mention that you think are important, let me know.

### CODE REVIEW

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -38,12 +38,86 @@ An easier solution would be to use Docker.
 1. Run `npm start` to start telescope.
 *If you get a series of errors, you may have to start redis-server depending on your installation configuration, do this by running the command `redis-server` in a seperate command window).*
 
-#### Logging
-The logger.js module in the utils folder of the repository exports a logger instance that can be imported in other files and used to log important events in production as well as help in debugging during development. Refer to [logging.md](logging.md) for more details.
+## Workflow in git and GitHub
```
[the remark this answers is not in the document]
Comment (humphd) [reply docs/CONTRIBUTING.md]: I think it's `git`, despite what other people on the internet think, they're wrong :)

**Comment by humphd** [comment]
OK, fixes in, and rebased on the new Prettier work, so some of the formatting is out of my hands.

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -45,15 +46,90 @@ Some helpful guides:
...
+   npm run eslint-fix
+   npm test
+   ...manually fix any errors yourself, rerunning npm test each time to confirm
+   ```
+1. When you're done, push to your fork, using the same name as your branch:
+   ```
+   git push origin issue-123
+   ```
+1. This will give you a link to GitHub to create a Pull Request.
+1. Create your Pull Request, and give it a title `Fix #123: short description of fix` (NOTE: the "Fix #123" part, which will [automatically close the issue](https://help.github.com/en/github/managing-your-work-on-github/closing-issues-using-keywords) when this gets merged) as well as a full description to talk about everything you did, any follow-up work that's necessary, how to test the code, etc.
```
Comment (manekenpix) [inline docs/CONTRIBUTING.md]: GitHub says [here](https://help.github.com/en/github/managing-your-work-on-github/closing-issues-using-keywords):
> "Closes #123" or "Closes: #123" in your pull request **description** or **commit message**.

I'm not sure if it applies to titles as well.

Comment (SeanPrashad) [reply docs/CONTRIBUTING.md]: Try it out with a test issue, test branch and test PR!! Lettuce know what happens

Comment (humphd) [reply docs/CONTRIBUTING.md]: Works in either place ime.

**Comment by humphd** [comment]
If there are any more comments on this, I'd like to get them so I can get this merged.

**Pull request merged**